AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioRecordingFieldController.kt

```diff
@@ -0,0 +1,120 @@
+import com.ichi2.ui.FixedTextView
+import com.ichi2.utils.KotlinCleanup
+import com.ichi2.utils.UiUtil.makeBold
+import java.io.File
+
+/**
+ * This controller always return a temporary path where it writes the audio
+ */
+class BasicAudioRecordingFieldController : FieldControllerBase(), IFieldController {
+
+    @KotlinCleanup("nun-null & lateinit")
+    private var mTempAudioPath: String? = null
+    @KotlinCleanup("nun-null & lateinit")
+    private var mAudioView: AudioView? = null
+
+    override fun createUI(context: Context, layout: LinearLayout?) {
+        val origAudioPath = mField.audioPath
+        var bExist = false
+        if (origAudioPath != null) {
+            val f = File(origAudioPath)
+            if (f.exists()) {
+                mTempAudioPath = f.absolutePath
+                bExist = true
+            }
+        }
+        if (!bExist) {
+            mTempAudioPath = generateTempAudioFile(mActivity)
+        }
+
+        // FIXME: We should move this outside the scrollview as it should always be on the screen.
+        mAudioView = createRecorderInstance(
+            context = mActivity,
+            resPlay = R.drawable.ic_play_arrow_white_24dp,
+            resPause = R.drawable.ic_pause_white_24dp,
+            resStop = R.drawable.ic_stop_white_24dp,
+            resRecord = R.drawable.ic_rec,
+            resRecordStop = R.drawable.ic_rec_stop,
+            audioPath = mTempAudioPath!!
+        )
+        mAudioView!!.setOnRecordingFinishEventListener(object : OnRecordingFinishEventListener {
+            override fun onRecordingFinish(v: View?) {
+                // currentFilePath.setText("Recording done, you can preview it. Hit save after finish");
+                // FIXME is this okay if it is still null?
+                mField.audioPath = mTempAudioPath
+                mField.setHasTemporaryMedia(true)
+            }
+        })
+        layout!!.addView(mAudioView, LinearLayout.LayoutParams.MATCH_PARENT)
+
+        context.apply {
+            // add preview of the field data to provide context to the user
+            // use a separate scrollview to ensure that the content does not push the buttons off-screen when scrolled
+            val sv = ScrollView(this)
+            layout.addView(sv)
+            val previewLayout = LinearLayout(this) // scrollView can only have one child
+            previewLayout.orientation = LinearLayout.VERTICAL
+            sv.addView(previewLayout)
+            val label = FixedTextView(this)
+            label.textSize = 20f
+            label.text = makeBold(this.getString(R.string.audio_recording_field_list))
+            label.gravity = Gravity.CENTER_HORIZONTAL
+            previewLayout.addView(label)
+            for (i in 0 until mNote.initialFieldCount) {
+                val field = mNote.getInitialField(i)
+                val textView = FixedTextView(this)
+                textView.text = field!!.text
```

Comment:
```suggestion
                textView.text = field?.text ?: ""
```

just a suggestion: I think it would be good if we use here an elvis, so that in any case we got null our app will not through the NPE.

Reply:
We could do with a `KotlinCleanup` here - I feel `getInitialField` shouldn't return a nullable

Reply:
> just a suggestion: I think it would be good if we use here an elvis, so that in any case we got null our app will not through the NPE.

Good one, i replaced `!!` by elvis.

> We could do with a `KotlinCleanup` here - I feel `getInitialField` shouldn't return a nullable

added `@KotlinCleanup` to the interface of `getInitialField`